utils/market_makers.py

```diff
@@ -28,16 +34,81 @@ def calculate_y(self, x):
   def calculate_spot_price(self):
     return 1
 
+  def define_sell_buy(self,typeTokenSell,balanceX,balanceY):
+    if typeTokenSell == 'X':
+      tokensData = {
+        'typeTokenBuy': 'Y',
+        'initialAmountSell': balanceX,
+        'initialAmountBuy': balanceY,
+      }
+      return tokensData
+    if typeTokenSell == 'Y': 
+      tokensData = {
+        'typeTokenBuy': 'X',
+        'initialAmountSell': balanceY,
+        'initialAmountBuy': balanceX,
+      }
+      return tokensData
+
+  def calculate_trade(self, initialAmountSell, initialAmountBuy, amountTokenSell):
+    k =  initialAmountSell + initialAmountBuy
+    amountTokenBuy = initialAmountSell + initialAmountBuy + amountTokenSell - k
+    price = amountTokenBuy/amountTokenSell
+    finalAmountSell = initialAmountSell+amountTokenSell
+    finalAmountBuy = initialAmountBuy-amountTokenBuy
+
+    transaction = {
+      'amountTokenBuy': amountTokenBuy,
+      'price': price,
+      'transactionSell': [initialAmountSell,finalAmountSell],
+      'transactionBuy': [initialAmountBuy,finalAmountBuy],
+      'label': ['Before the trade', 'After the trade']
+    }
+
+    return transaction
+
 
 class Uniswap(MarketMaker):
   def get_constant(self, x, y):
-    return x+y
+    return x*y
   
   def calculate_y(self, x):
     return self.constant / x
   
   def calculate_spot_price(self, x):
     return self.constant / (x**2)
+  
+  def define_sell_buy(self,typeTokenSell,balanceX,balanceY):
+    if typeTokenSell == 'X':
+      tokensData = {
+        'typeTokenBuy': 'Y',
+        'initialAmountSell': balanceX,
+        'initialAmountBuy': balanceY,
+      }
+      return tokensData
+    if typeTokenSell == 'Y': 
```

Comment:
Since we are working with binary pools, I think that would be better to the typeTokenSell to be a binary variable. Also, you don't need the second if statement.

Reply:
With the if statement I agree, but with the binary variable I'm not sure, because I use this variable to render text as well, if it was a binary variable I would have to add logic into the streamlit code. Do you this is still worth it?

Reply:
if part done on commit [0dd4bda78c680f4c3037b3703c6be982b60878b8](https://github.com/luizakp/streamlittest/pull/2/commits/0dd4bda78c680f4c3037b3703c6be982b60878b8)

Reply:
You don't need the else statement either.

Reply:
I see, makes sense. However, it seems a little hard-coded, but it is ok for now. In the future, it would make sense to redefine the class attributes to cover this. For example, passing the input and output token names in the constructor.

Reply:
done on commit [9cb231779766329df96881769d393bd5a7efb74a](https://github.com/luizakp/streamlittest/pull/2/commits/9cb231779766329df96881769d393bd5a7efb74a)